### PR TITLE
ERC20 pricing

### LINKED ISSUE #194: Allow ERC20/827 tokens as Listing prices
Listings should be able to specify the token they want to be paid in. 

This will allow for better interaction with parters, and allow quasi-fiat pricing with stable coins like DAI. 

I began work in branch `topic/erc20_pricing`: https://github.com/OriginProtocol/origin-js/tree/topic/erc20_pricing

(Not tested yet as I can't get `npm install` to work on the plane!)

### PR DESCRIPTION
**Pull request by tyleryasaka**
### Checklist:

- [x] Code contains relevant tests for the problem you are solving
- [x] Ensure all new and existing tests pass
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/docs)
- [x] Submit to the `develop` branch instead of `master`

### Description:

Updates contracts and API to allow listings and purchases to be done in an ERC20 token, which is specified on listing creation.

We don't have any UI designs so I'm just considering this experimental for now - however, I wanted to merge as this will make it easier to begin supporting ERC20 tokens in the Dapp when we want to do that.

Closes #194

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,10 @@
     "cross-fetch": "^2.1.1",
     "ethereumjs-util": "^5.2.0",
     "form-data": "^2.3.2",
+    "ipfs-api": "^20.2.0",
```
Comment (nick) [inline package.json]: we need this again?

Comment (tyleryasaka) [reply package.json]: hmm no... must have messed up the merge

Comment (tyleryasaka) [reply package.json]: Removed

**Comment by wanderingstan** [comment]
This never got merged. @tyleryasaka , should we try to get this in? Seems like low-hanging fruit, though of course might be part of a bigger token discussion.

**Comment by tyleryasaka** [comment]
@wanderingstan I think it's pretty out of date now. I don't think this was stated as a Q3 deliverable so I'm thinking we should hold off for now.

**Pull request closed without merging**